ledger/src/blockstore.rs

```diff
@@ -170,6 +172,48 @@ impl<T> AsRef<T> for WorkingEntry<T> {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct LastFECSetCheckResults {
+    is_full: bool,
+    is_retransmitter_signed: bool,
+}
+
+impl LastFECSetCheckResults {
+    pub fn report_metrics(&self, slot: Slot, hash: Hash) {
+        if !self.is_full {
+            datapoint_warn!(
+                "incomplete_final_fec_set",
+                ("slot", slot, i64),
+                ("hash", hash.to_string(), String)
+            );
+        }
+        if !self.is_retransmitter_signed {
+            datapoint_warn!(
+                "invalid_retransmitter_signature_final_fec_set",
+                ("slot", slot, i64),
+                ("hash", hash.to_string(), String)
```

Comment:
Apparently these `hash` things are pretty expensive for metrics server because they do not compress.
And until chained merkle shreds are rolled out every node is going to submit this metric for every slot which is too much.

Reply:
got it, my understanding is that we want  the metrics to verify the check is working before we turn on the feature flag.
Would it be preferrable to instead gate these metrics by:
* `standard_broadcast_run::should_chain_merkle_shreds`
* `feature_set::verify_retransmitter_signature`

so that we can verify before turning on:

* `feature_set::vote_only_full_fec_sets`
* `feature_set::vote_only_retransmitter_signed` (new flag I will create as per https://github.com/anza-xyz/agave/pull/2101#discussion_r1679387761)

Reply:
Sounds good i've gated this with `should_chain_merkle_shreds` so that we don't wastefully submit this metric prematurely.

Reply:
But then the metric cannot be used to evaluate if we are ready to activate the feature or not because the metric itself is behind the feature.

I think for now there is no point in adding the metric because it will spam metrics server until chained Merkle shreds are rolled out.
Putting it behind the feature gate is not helpful either because then we don't know if we can activate the feature or not.

So I think the best is to remove it entirely for now, wait until chained Merkle shreds are fully rolled out, then we can add this metric to confirm the cluster is ready to activate duplicate shreds features.

Reply:
i thought the `should_chain_merkle_shreds` function is the one that controls the chained merkle rollout?
As in, you will slowly update this function to target different cluster types and increasing slot ranges until it is always `true`.
By gating this metric behind `should_chain_merkle_shreds` we can verify if it is performing correctly on chained merkle shred slots only, and once `should_chain_merkle_shreds` is always true we can turn on `feature_set::vote_only_retransmitter_signed_fec_sets`

Reply:
I suppose there would be some metrics spam as the cluster upgrades to a version that changes `should_chain_merkle_shreds` as there would be disagreement on whether this slot should use the new shred format.
What do you think about using `feature_set::verify_retransmitter_signature::id()` to gate the metric instead? since I assume we will activate that feature flag first (after chained merkle roots are fully rolled out) but before turning on this one. 
If not, I can do as you suggest and remove the metric entirely and add it back before we intend to activate the feature.